### PR TITLE
test: Cypess - added tags - @tag.Fork, @tag.Git

### PR DESCRIPTION
**Pull request by sharanya-appsmith**
- cypress/e2e/Regression/ClientSide/Fork
- client/cypress/e2e/Regression/ClientSide/Git

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

#### How Has This Been Tested?
- [x] Manual
- [x] Cypress
>
>

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Refactor**
  - Improved the structure and readability of test suites related to forking applications.

- **Tests**
  - Enhanced test descriptions and categorization with new tags for better test management.
  - Added new test scenarios for forking applications by non-signed users and checking application properties post-fork.

- **Chores**
  - Introduced new tags for test suites to facilitate automated test height adjustment and forking feature identification.

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/Fork/ForkApplicationReconnectModal_spec.ts [hunk 1]
```diff
@@ -6,44 +6,48 @@ import {
...
-    homePage.DeleteWorkspace(forkWorkspaceName);
-  });
-});
+
+    after(() => {
+      homePage.NavigateToHome();
+      homePage.DeleteApplication(currentAppName);
+      homePage.DeleteApplication(currentAppName);
+      homePage.DeleteWorkspace(currentWorkspace);
+      homePage.DeleteWorkspace(forkWorkspaceName);
```
[the remark this answers is not in the document]
Comment (mohanarpit) [reply app/client/cypress/e2e/Regression/ClientSide/Fork/ForkApplicationReconnectModal_spec.ts:50]: @sharanya-appsmith I know that we are working on tags, but I'm thinking if we should fix this comment. What do you think? How is this test case working today?

File: app/client/cypress/tags.js [hunk 2]
```diff
@@ -32,5 +32,7 @@ module.exports = {
     "@tag.Theme",
     "@tag.Settings",
     "@tag.ImportExport",
+    "@tag.AutoHeight",
```
Comment (mohanarpit) [inline app/client/cypress/tags.js:35]: @sharanya-appsmith Minor comment: For now, I'm OK with this. We are adding AutoHeight tag here but the PR to tag the Cypress tests is in a separate PR. Can we please ensure that we make these changes consistently?

File: app/client/cypress/e2e/Regression/ClientSide/Fork/ForkApplicationReconnectModal_spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sharanya-appsmith) [reply app/client/cypress/e2e/Regression/ClientSide/Fork/ForkApplicationReconnectModal_spec.ts:50]: The test case forks the application. So there are 2 apps in the same name ( 1 app created, 1 app forked). Hence it calling DeleteApplication twice with the same app name works.

File: app/client/cypress/tags.js (hunk 2 above)
Comment (sharanya-appsmith) [reply app/client/cypress/tags.js:35]: Yes. understood. will take care of this hereon.

**Review by mohanarpit — APPROVED** [review]

**Pull request merged**